### PR TITLE
Replace RegexpDocumentParser with LookaheadParser at NagFortranParser

### PR DESCRIPTION
**Pull request by kaschel-hm**

### CODE REVIEW

**Comment by uhafner** [comment]
Do you need more details for that PR?

**Comment by kaschel-hm** [comment]
Excuse my late reply. The only problem was the multiline errors, because there is no indicator (e.g. a \n) in the lines that the next line belongs to it. My approach is to check whether the next line is indented accordingly. I hope that's right.

File: src/main/java/edu/hm/hafner/analysis/parser/NagFortranParser.java [hunk 1]
```diff
@@ -41,6 +44,27 @@ public NagFortranParser() {
...
+            StringBuilder messageBuilder = new StringBuilder(matcher.group(5));
+            String category = matcher.group(1);
+
+            while (lookahead.hasNext("\\s+ .+")) {
+                messageBuilder.append("\n");
+                messageBuilder.append(lookahead.next());
+            }
+
+            return builder.setFileName(matcher.group(2))
+                    .setLineStart(getLineNumber(matcher))
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/NagFortranParser.java]: replace with `parseInt(matcher.group(4))` and remove `getLineNumber` method

Comment (kaschel-hm) [reply src/main/java/edu/hm/hafner/analysis/parser/NagFortranParser.java]: The test data contains entries that do not have a line number (`NagFortranError.txt`), so `parseInt` would throw an `NumberFormatException`. Is the test data valid or must it be modified?

Comment (uhafner) [reply src/main/java/edu/hm/hafner/analysis/parser/NagFortranParser.java]: Ah: `IntegerParser.parseInt` does handle the exception already, it is used by all other parsers as well.

**Pull request merged**